src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -148,6 +166,71 @@ private <T> List<T> loadData(final String configKey, Type type)
 		}
 	}
 
+	private List<InventorySetup> loadV1Setups() {
+		Type setupTypeV1 = new TypeToken<ArrayList<InventorySetup>>()
+		{
+
+		}.getType();
+		return loadData(CONFIG_KEY_SETUPS, setupTypeV1);
+	}
+
+	private List<InventorySetup> loadV2Setups() {
+		Type setupTypeV2 = new TypeToken<ArrayList<InventorySetupSerializable>>()
+		{
+
+		}.getType();
+		List<InventorySetupSerializable> issList = new ArrayList<>(loadData(CONFIG_KEY_SETUPS_V2, setupTypeV2));
+		List<InventorySetup> isList = new ArrayList<>();
+		for (final InventorySetupSerializable iss : issList)
+		{
+			isList.add(InventorySetupSerializable.convertToInventorySetup(iss));
+		}
+		return isList;
+	}
+
+	private InventorySetup loadV3Setup(String configKey) {
+		final String storedData = configManager.getConfiguration(CONFIG_GROUP, configKey);
+		try
+		{
+			return InventorySetupSerializable.convertToInventorySetup(gson.fromJson(storedData, InventorySetupSerializable.class));
+		}
+		catch (Exception e)
+		{
+			log.error(String.format("Exception occurred while loading %s", configKey), e);
+			throw e;
+		}
+	}
+
+	private List<InventorySetup> loadV3Setups() {
+		final String wholePrefix = ConfigManager.getWholeKey(CONFIG_GROUP, null, CONFIG_KEY_SETUPS_V3_PREFIX);
+		final List<String> loadedSetupWholeKeys = configManager.getConfigurationKeys(wholePrefix);
+		Set<String> loadedSetupKeys = loadedSetupWholeKeys.stream().map(
+			key -> key.substring(wholePrefix.length()-CONFIG_KEY_SETUPS_V3_PREFIX.length())
+		).collect(Collectors.toSet());
+
+		Type setupsOrderType = new TypeToken<ArrayList<String>>()
+		{
+
+		}.getType();
+		final String setupsOrderJson = configManager.getConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_ORDER_V3);
+		final List<String> setupsOrder = gson.fromJson(setupsOrderJson, setupsOrderType);
+
+		List<InventorySetup> loadedSetups = new ArrayList<>();
+		for (final String configHash : setupsOrder)
+		{
+			final String configKey = CONFIG_KEY_SETUPS_V3_PREFIX + configHash;
+			if (loadedSetupKeys.remove(configKey)) { // Handles if hash is present only in configOrder.
+				loadedSetups.add(loadV3Setup(configKey));
+			}
+		}
+		for (final String configKey : loadedSetupKeys)
+		{
+			// Load any remaining setups not present in setupsOrder. Useful if updateConfig crashes midway.
+			loadedSetups.add(loadV3Setup(configKey));

```

Comment:
move the inner function call to the line before.

Reply:
done